src/main/java/com/mathworks/ci/RunMatlabTestsStep.java

```diff
@@ -0,0 +1,185 @@
+    }
+
+
+    @Override
+    public StepExecution start(StepContext context) throws Exception {
+        Launcher launcher = context.get(Launcher.class);
+        FilePath workspace = context.get(FilePath.class);
+        
+        //Copy Scratch file needed to run MATLAB tests in workspace
+        FilePath targetWorkspace = new FilePath(launcher.getChannel(), workspace.getRemote());
+        copyScratchFileInWorkspace(MatlabBuilderConstants.MATLAB_TESTS_RUNNER_RESOURCE,
+                MatlabBuilderConstants.MATLAB_TESTS_RUNNER_TARGET_FILE, targetWorkspace);
+        return new MatlabStepExecution(context,constructCommandForTest(getInputArgs()));
+    }
+    
+    @Extension
+    public static class CommandStepDescriptor extends StepDescriptor {
+
+        @Override
+        public Set<? extends Class<?>> getRequiredContext() {
+            return ImmutableSet.of(TaskListener.class, FilePath.class, Launcher.class,
+                    EnvVars.class, Run.class);
+        }
+
+        @Override
+        public String getFunctionName() {
+            return Message.getValue("matlab.tests.build.step.name");
+        }
+    }
+    
+    public String constructCommandForTest(String inputArguments) {
+        final String matlabFunctionName =
+                FilenameUtils.removeExtension(MatlabBuilderConstants.MATLAB_TESTS_RUNNER_TARGET_FILE);
+        final String runCommand = "exit(" + matlabFunctionName + "(" + inputArguments + "))";
+        return runCommand;
+    }
+
+    
+    private String getInputArgs() {
+        final List<String> inputArgs = new ArrayList<>();
+        final Map<String, String> args = getMatlabArgs();
+
+        args.forEach((key, val) -> {
+            if (val != null) {
+                inputArgs.add("'" + key + "'" + "," + "'" + val.replaceAll("'", "''") + "'");
+            }
+        });
+
+        if (inputArgs.isEmpty()) {
+            return "";
+        }
+
+        return String.join(",", inputArgs);
+    }
+    
+    private Map<String, String> getMatlabArgs() {
+        final Map<String, String> args = new HashMap<String, String>();
+        args.put(PDF_REPORT_PATH,getTestResultsPDF());
+        args.put(TAP_RESULTS_PATH,getTestResultsTAP());
+        args.put(JUNIT_RESULTS_PATH,getTestResultsJUnit());
+        args.put(STM_RESULTS_PATH, getTestResultsSimulinkTest());
+        args.put(COBERTURA_CODE_COVERAGE_PATH, getCodeCoverageCobertura());
+        args.put(COBERTURA_MODEL_COVERAGE_PATH, getModelCoverageCobertura());
+        return args;
+    }
+    
+    /*
+     * Method to copy given file from source to target node specific workspace.
+     */
+    private void copyScratchFileInWorkspace(String sourceFile, String targetFile, FilePath targetWorkspace)
+            throws IOException, InterruptedException {
+        final ClassLoader classLoader = getClass().getClassLoader();
+        FilePath targetFilePath = new FilePath(targetWorkspace, targetFile);
```

Comment:
minor nitpick. this could move down one line closer to first use.

Reply:
Done.